packages/x-components/src/components/modals/base-modal.vue

```diff
@@ -87,6 +100,40 @@
       if (this.open) {
         this.syncBody(true);
       }
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      const resizeObserver = new ResizeObserver(this.updatePosition);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      this.$watch(
+        'clientHeaderSelector',
+        () => {
+          if (this.clientHeaderSelector) {
+            const element = document.querySelector(this.clientHeaderSelector);
+            if (element) {
+              this.selectedElement = element;
+              resizeObserver.observe(element);
+            }
+          }
+        },
+        { immediate: true }
+      );
+    }
+
+    /**
+     * Updates the position of the modal setting the top of the element depending
+     * on the selector. The modal will be placed under this selector.
+     *
+     * @internal
+     */
+    @Debounce(100, { leading: true })
+    updatePosition(): void {
+      this.$refs.modal.style.top = this.clientHeaderSelector
+        ? `${
+            this.selectedElement.getBoundingClientRect().height +
+            this.selectedElement.getBoundingClientRect().y
+          }px`
+        : '0';
```

Comment:
Regarding this implementation. We are only allowing to position an element below another one. Should we make this positioning also configurable? I don't know what exactly the purpose of the task was, but it could be something to consider. On a side note, I think we should check if the `referenceElement` exists, rather than if a `referenceSelector` has been defined. Maybe there's a `referenceSelector` defined but no element matches it in the DOM, resulting in a `Reference error` due to `referenceSelector` not being defined.

Looking at the code, calling `getBoundingClientRect()` method causes a layout `reflow`. We are calling twice consecutively, which I assume are `batched` and result in a single call. However, there are other approaches to avoid calling `getBoundingClientRect()` or to reduce its `reflow` penalty. For further information regarding alternatives, you can check this [post](https://toruskit.com/blog/how-to-get-element-bounds-without-reflow/). I don't think we have to refactor it now, but it might be worth to consider it, at least for the future.

```suggestion
      const { height, y } = this.selectedElement?.getBoundingClientRect() ?? { height: 0, y: 0 };
      this.$refs.modal.style.top = `${height + y}px`;
```

Reply:
At this point we want to position the modal under the selected element, maybe in the future, we can add the other logic.

About the other point, I am not sure if its something that we have to be worried about